src/test/scripts/check_enterprise_merge.sh

```diff
@@ -1,55 +1,97 @@
 git config user.name "citus bot"
 
-# reset repository into usable state if script ran before
-git fetch origin
-git reset --hard
-git checkout enterprise-master
-git reset --hard origin/enterprise-master
-
-branch_name="${CIRCLE_BRANCH}"
-
-# check if the branch on community exists on enterprise
-# the output will not be empty if it does
-if [ `git branch -r --list origin/$branch_name` ]
-then
-    try_merge enterprise-master origin/$branch_name
-else
-    # add community as a remote if not already added
-    set +e
-    if ! git ls-remote community > /dev/null; then
-        set -e
-        git remote add --no-tags community git@github.com:citusdata/citus.git
-    fi
-    set -e
-
-    # prevent pushes to community and update branch we care about
-    git remote set-url --push community no-pushing
-    git fetch community $branch_name
-
-    try_merge enterprise-master community/$branch_name
+# Disable "set -x" again, because $ENTERPRISE_REMOTE contains passwords
+{ set +x ; } 2> /dev/null
+git remote add enterprise "$ENTERPRISE_REMOTE"
+set -x
+
+git remote set-url --push enterprise no-pushing
+
+# Fetch enterprise-master
+git fetch enterprise enterprise-master
+
+
+# Try to fetch the enterprise version of the branch. If it fails we assume it
+# does not exist. This seems reasonble, because even in case of other failures
+# (e.g. network) we still continue as if the enterprise version of the branch
+# does not exist.
+if ! git fetch enterprise "$PR_BRANCH" > /dev/null 2>&1 ; then
+    echo "INFO: enterprise/$PR_BRANCH was not found"
+    # If the current branch does not exist on the enterprise repo, then all we
+    # have to check is if it can be merged into enterprise master without
+    # problems.
+    # this will exit since -e option is set and it will return non-zero code on conflicts.
+    git checkout enterprise/enterprise-master
+    # Check if we can merge the PR branch into enterprise-master
+    git merge --no-ff --no-commit "origin/$PR_BRANCH"
+    exit 0
+fi
+
+# Show the top commit of the enterprise branch to make debugging easier
+git log -n 1 "enterprise/$PR_BRANCH"
+
+# Check that this branch contains the top commit of the enterprise-master
+# branch. If it does not it means it might not be able to be merged into it
+# automatically. So we fail in that case.
+if ! git merge-base --is-ancestor enterprise/enterprise-master "enterprise/$PR_BRANCH" ; then
+    echo "ERROR: enterprise/$PR_BRANCH is not up to date with enterprise-master"
+    exit 1
 fi
+
+# Check that this branch contains the top commit of the current community PR
+# branch. If it does not it means it's not up to date with the current PR, so
+# the enterprise branch should be updated.
+if ! git merge-base --is-ancestor "origin/$PR_BRANCH" "enterprise/$PR_BRANCH" ; then
+    echo "ERROR: enterprise/$PR_BRANCH is not up to date with community PR branch"
+    exit 1
+fi
+
+# Because of the two checks above we now know that it contains both the last
```

Comment:
This part is implied by the if checks I guess, and it makes sense but for explicitly showing that there is no conflict, it could be nice to use the previous `try_merge` method here. It is not going to make the job any slower, and it makes the script easier to understand IMHO.